app/src/main/java/com/flab/findmycat/ui/list/CatListAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.flab.findmycat.ui.list
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.lifecycle.LifecycleOwner
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.flab.findmycat.databinding.CatItemBinding
+import com.flab.findmycat.domain.Cat
+
+class CatClickListener(val clickListener: (id: String) -> Unit) {
+    fun onClick(cat: Cat) = clickListener(cat.id ?: "")
+}
+
+class CatListAdapter(private val lifecycleOwner: LifecycleOwner, private val clickListener: CatClickListener) :
+    ListAdapter<Cat, CatListAdapter.CatListViewHolder>(DiffCallback) {
+
+    class CatListViewHolder(
+        private var binding: CatItemBinding,
+        private val lifecycleOwner: LifecycleOwner,
+        private val clickListener: CatClickListener
+    ) : RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(cat: Cat) {
+            binding.lifecycleOwner = lifecycleOwner
+            binding.clickListener = clickListener
+            binding.cat = cat
+        }
+    }
+
+    companion object DiffCallback : DiffUtil.ItemCallback<Cat>() {
+        override fun areItemsTheSame(oldItem: Cat, newItem: Cat): Boolean {
+            return oldItem.id == newItem.id
+        }
+
+        override fun areContentsTheSame(oldItem: Cat, newItem: Cat): Boolean {
+            return oldItem.name == newItem.name
```

Comment:
```suggestion
            return oldItem.name == newItem.name
```
`areContentsTheSame`을 이렇게 구현하면 발생할수있는 문제점에 대해서 알고계신가요? 만약 Diff 대상 아이템 (old, new)의 name값은 동일한데 image가 다른경우 어떻게 동작할까요?

Reply:
같은 값이라고 인식하고 화면에 보여주지 않습니다. 혹시 모르니 image값도 비교해야겠네요!

Reply:
data class CatViewHolderUiModel(val id: String, val image: String), ==